packages/components/src/internal/query/api.ts

```diff
@@ -934,31 +921,106 @@ export function deleteRows(options: DeleteRowsOptions): Promise<DeleteRowsRespon
             queryName: schemaQuery.queryName,
             success: response => {
                 resolve(
-                    Object.assign(
-                        {},
-                        {
-                            schemaQuery: options.schemaQuery,
-                            rows: response.rows,
-                            transactionAuditId: response.transactionAuditId,
-                        }
-                    )
+                    new QueryCommandResponse({
+                        schemaQuery: options.schemaQuery,
+                        rows: response.rows,
+                        transactionAuditId: response.transactionAuditId,
+                    })
                 );
             },
             failure: error => {
-                reject(
-                    Object.assign(
-                        {},
-                        {
-                            schemaQuery: options.schemaQuery,
-                        },
-                        error
-                    )
-                );
+                reject({
+                    schemaQuery: options.schemaQuery,
+                    error,
+                });
             },
         });
     });
 }
 
+export function saveRows(options: Query.SaveRowsOptions): Promise<Query.SaveRowsResponse> {
+    return new Promise((resolve, reject) => {
+        Query.saveRows({
+            apiVersion: 13.2,
+            ...options,
+            success: response => {
+                resolve(response);
+            },
+            failure: error => {
+                reject(error);
+            },
+        });
+    });
+}
+
+export function deleteRowsByContainer(
+    options: DeleteRowsOptions,
+    containerField: string = 'ContainerPath'
+): Promise<QueryCommandResponse> {
+    const commands = [];
+
+    const allRows = options.rows;
+    if (
+        !allRows ||
+        Object.keys(allRows[0])
+            .map(key => key.toLowerCase())
+            .indexOf(containerField.toLowerCase()) === -1
+    )
+        return deleteRows(options);
```

Comment:
I'm a little confused by this block. If the options don't have rows, won't the deleteRows API call fail? Is this just calling through to deleteRows in that case so that the error handling is the same / consistent?

Reply:
Yes this is to fall back to the default behavior of empty rows for deleteRows, it's unlikely to occur.